test/non_ssl.jl

```diff
@@ -0,0 +1,308 @@
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: Binary.",
+        "213 ...",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete.",
+    ]
+    test_get(headers, options)
+
+    # download a file to a specific path
+    headers = [
+        "220 pyftpdlib ... ready.",
+        "331 Username ok, send password.",
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: Binary.",
+        "213 ...",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete."
+        ]
+
+    save_path = joinpath(pwd(), "test_path.txt")
+    @test !isfile(save_path)
+
+    local_file = download_file
+
+    resp = ftp_get(options, local_file, save_path)
+    server_file = joinpath(ROOT, local_file)
+    body = readstring(server_file)
+
+    test_response(resp, 226, headers, save_path, body)
+
+    rm(save_path)
+
+    # ftp connect
+    headers = [
+        "220 pyftpdlib ... ready.",
+        "331 Username ok, send password.",
+        "230 Login successful.",
+        "257 \"/\" is the current directory.",
+        mode_header,
+        "200 Type set to: ASCII.",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete.",
+    ]
+
+    ctxt, resp = ftp_connect(options)
+    directory_name = "test_directory"
+    body = readstring(resp.body)
+    @test contains(body, download_file)
+    @test contains(body, directory_name)
+    @test resp.bytes_recd == length(body)
+    @test resp.code == 226
+    @test is_headers_equal(resp.headers, headers)
+    @test ctxt.url == options.url == "ftp://$(hostname(server))/"
+    @test ctxt.options == options
+    ftp_close_connection(ctxt)
+
+
+    headers = [
+        mode_header,
+        "257 \"/\" is the current directory."
+    ]
+    ctxt, resp = ftp_connect(options)
+    test_command(headers, ctxt)
+
+    @test ctxt.url == options.url == "ftp://$(hostname(server))/"
+    @test ctxt.options == options
+    ftp_close_connection(ctxt)
+
+    headers = [
+        mode_header,
+        "200 Type set to: Binary.",
+        "213 ...",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete.",
+    ]
+    ctxt, resp = ftp_connect(options)
+    test_get(headers, ctxt)
+
+    @test ctxt.url == options.url == "ftp://$(hostname(server))/"
+    @test ctxt.options == options
+    ftp_close_connection(ctxt)
+
+    headers = [
+        mode_header,
+        "200 Type set to: Binary.",
+        "125 Data connection already open. Transfer starting.",
+        "226 Transfer complete."
+        ]
+
+    ctxt, resp = ftp_connect(options)
+    test_put(headers, ctxt)
+
+end
+
+tests_by_mode(true)
+tests_by_mode(false)
+
+# test binary vs ascii
+@unix_only upload_bytes = string("466F6F426172", "0A", "466F6F426172")
```

Comment:
`@unix_only` is deprecated (I forgot). Switch to `is_unix() && (upload_bytes = ...)`